### PR TITLE
[Improve][Connector-v2][Jdbc] Refactor AbstractJdbcCatalog

### PR DESCRIPTION
**Pull request by whhe**
## Purpose of this pull request

Extract the common logic of MySqlCatalog and SqlServerCatalog to AbstractJdbcCatalog.

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Comment by TyrantLucifer** [comment]
Please add unit test cases.

**Comment by whhe** [comment]
> Please add unit test cases.

Seems SeaTunnel does not depend on Mockito now, I want to introduce it to mock jdbc connection, is it OK ? And if so what version should I choose?

**Comment by liugddx** [comment]
> Mockito

Refer to `seatunnel-e2e/connector-jdbc-e2e`

**Comment by whhe** [comment]
> > Mockito
> 
> Refer to `seatunnel-e2e/connector-jdbc-e2e`

So I should add IT cases in module connector-jdbc-e2e instead of UT cases in the modified module, am I right?

**Comment by EricJoy2048** [comment]
> > > Mockito
> > 
> > 
> > Refer to `seatunnel-e2e/connector-jdbc-e2e`
> 
> So I should add IT cases in module connector-jdbc-e2e instead of UT cases in the modified module, am I right?

Yes, you can add IT for this pr.

**Comment by ic4y** [comment]
@whhe Great commit! Now we just need to resolve the conflicts.

**Comment by whhe** [comment]
I rebased the code and seems it's coverd by the IT cases for now. @ic4y @EricJoy2048 @liugddx

**Comment by ic4y** [comment]
@XiaoJiang521 PTAL

File: seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/catalog/sqlserver/SqlServerCatalog.java [hunk 1]
```diff
@@ -292,99 +184,27 @@ private SeaTunnelDataType<?> fromJdbcType(String typeName, int precision, int sc
...
-            return ps.execute();
-        } catch (Exception e) {
-            throw new CatalogException(
-                    String.format(
-                            "Failed creating database %s in catalog %s",
-                            databaseName, this.catalogName),
-                    e);
-        }
+    protected String getCreateDatabaseSql(String databaseName) {
+        return String.format("CREATE DATABASE `%s`", databaseName);
```
Comment (XiaoJiang521) [inline seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/catalog/sqlserver/SqlServerCatalog.java]: sqlserver is CREATE DATABASE [%s]. no ``

File: seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/catalog/sqlserver/SqlServerCatalog.java [hunk 2]
```diff
@@ -292,99 +184,27 @@ private SeaTunnelDataType<?> fromJdbcType(String typeName, int precision, int sc
...
-            return ps.execute();
-        } catch (Exception e) {
-            throw new CatalogException(
-                    String.format(
-                            "Failed dropping database %s in catalog %s",
-                            databaseName, this.catalogName),
-                    e);
-        }
+    protected String getDropDatabaseSql(String databaseName) {
+        return String.format("DROP DATABASE IF EXISTS `%s`;", databaseName);
```
Comment (XiaoJiang521) [inline seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/catalog/sqlserver/SqlServerCatalog.java]: sqlserver is [%s]. no ``

File: seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/catalog/sqlserver/SqlServerCatalog.java (hunk 1 above)
Comment (whhe) [reply seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/catalog/sqlserver/SqlServerCatalog.java]: I just copied the sql templates from https://github.com/apache/seatunnel/blob/dev/seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/catalog/sqlserver/SqlServerCatalog.java

Will fix it later.

Comment (XiaoJiang521) [reply seatunnel-connectors-v2/connector-jdbc/src/main/java/org/apache/seatunnel/connectors/seatunnel/jdbc/catalog/sqlserver/SqlServerCatalog.java]: Yes, this is a historical bug. Thank you for your effort.

**Comment by whhe** [comment]
The failed case `JdbcPhoenixIT` seems not related to this pr.

**Comment by whhe** [comment]
@XiaoJiang521 Comments addressed, what do you think for now.

**Comment by XiaoJiang521** [comment]
LGTM

**Review by XiaoJiang521 — APPROVED** [review]

**Comment by whhe** [comment]
@ic4y @EricJoy2048 PTAL

**Comment by ic4y** [comment]
At present, no tests cover these modifications, and the UT tests are also not working. We should add a seatunnel-catalog-e2e module under the seatunnel-e2e module and use testcontainers to cover these tests.

**Comment by whhe** [comment]
> At present, no tests cover these modifications, and the UT tests are also not working. We should add a seatunnel-catalog-e2e module under the seatunnel-e2e module and use testcontainers to cover these tests.

Finally got caught with this work again ... will try it later.

**Comment by whhe** [comment]
I add `testCatalog` method in `AbstractJdbcIT` and enable it for MySql, Oracle and SqlServer in this pr, looks like these IT cases finally passed.

As PG does not use the same base IT class, I want to modify it in another pr, what do you think? @ic4y

**Comment by whhe** [comment]
Thanks to help from @EricJoy2048, CI jobs are totally green now.

**Comment by whhe** [comment]
All catalog implementations related to `AbstractJdbcCatalog` are covered by IT cases now, and UT failure is not related.

PTAL  @ic4y @liugddx @EricJoy2048

**Comment by liugddx** [comment]
> All catalog implementations related to `AbstractJdbcCatalog` are covered by IT cases now, and UT failure is not related.
> 
> PTAL @ic4y @liugddx @EricJoy2048

Waiting for #5403 to be merged

**Comment by whhe** [comment]
@liugddx Could you please help to retry the failed ut case? Seems the `testJobLevelCheckpointTimeOut` is not completely fixed.

**Comment by liugddx** [comment]
> @liugddx Could you please help to retry the failed ut case? Seems the `testJobLevelCheckpointTimeOut` is not completely fixed.

Yes, I found the problem, it's not very stable. I will fix this problem in other PR #5414

**Comment by whhe** [comment]
All tests passed, PTAL @ic4y @liugddx @EricJoy2048

**Comment by ic4y** [comment]
@XiaoJiang521 is familiar with this area, could you please review it?

**Comment by XiaoJiang521** [comment]
LGTM @ic4y

**Review by ic4y — APPROVED** [review]

**Review by EricJoy2048 — APPROVED** [review]

**Pull request merged**